src/celeritas/ext/detail/GeantSurfacePhysicsLoader.cc

```diff
@@ -0,0 +1,427 @@
+
+    if (auto iter = names.find(value); iter != names.end())
+    {
+        return names.find(value)->second;
+    }
+    return "UNKNOWN";
+
+#undef GSPL_OSM_PAIR
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Get a string corresponding to the \c G4OpticalSurfaceFinish selection.
+ */
+char const* to_cstring(G4OpticalSurfaceFinish value)
+{
+#define GSPL_OSF_PAIR(ENUMVALUE)                      \
+    {                                                 \
+        G4OpticalSurfaceFinish::ENUMVALUE, #ENUMVALUE \
+    }
+
+    static std::unordered_map<G4OpticalSurfaceFinish, const char*> const names
+        = {
+            GSPL_OSF_PAIR(polished),
+            GSPL_OSF_PAIR(polishedfrontpainted),
+            GSPL_OSF_PAIR(polishedbackpainted),
+
+            GSPL_OSF_PAIR(ground),
+            GSPL_OSF_PAIR(groundfrontpainted),
+            GSPL_OSF_PAIR(groundbackpainted),
+
+            GSPL_OSF_PAIR(polishedlumirrorair),
+            GSPL_OSF_PAIR(polishedlumirrorglue),
+            GSPL_OSF_PAIR(polishedair),
+            GSPL_OSF_PAIR(polishedteflonair),
+            GSPL_OSF_PAIR(polishedtioair),
+            GSPL_OSF_PAIR(polishedtyvekair),
+            GSPL_OSF_PAIR(polishedvm2000air),
+            GSPL_OSF_PAIR(polishedvm2000glue),
+
+            GSPL_OSF_PAIR(etchedlumirrorair),
+            GSPL_OSF_PAIR(etchedlumirrorglue),
+            GSPL_OSF_PAIR(etchedair),
+            GSPL_OSF_PAIR(etchedteflonair),
+            GSPL_OSF_PAIR(etchedtioair),
+            GSPL_OSF_PAIR(etchedtyvekair),
+            GSPL_OSF_PAIR(etchedvm2000air),
+            GSPL_OSF_PAIR(etchedvm2000glue),
+
+            GSPL_OSF_PAIR(groundlumirrorair),
+            GSPL_OSF_PAIR(groundlumirrorglue),
+            GSPL_OSF_PAIR(groundair),
+            GSPL_OSF_PAIR(groundteflonair),
+            GSPL_OSF_PAIR(groundtioair),
+            GSPL_OSF_PAIR(groundtyvekair),
+            GSPL_OSF_PAIR(groundvm2000air),
+            GSPL_OSF_PAIR(groundvm2000glue),
+
+            GSPL_OSF_PAIR(Rough_LUT),
+            GSPL_OSF_PAIR(RoughTeflon_LUT),
+            GSPL_OSF_PAIR(RoughESR_LUT),
+            GSPL_OSF_PAIR(RoughESRGrease_LUT),
+
+            GSPL_OSF_PAIR(Polished_LUT),
+            GSPL_OSF_PAIR(PolishedTeflon_LUT),
+            GSPL_OSF_PAIR(PolishedESR_LUT),
+            GSPL_OSF_PAIR(PolishedESRGrease_LUT),
+
+            GSPL_OSF_PAIR(Detector_LUT),
+        };
+
+    if (auto iter = names.find(value); iter != names.end())
+    {
+        return names.find(value)->second;
+    }
+    return "UNKNOWN";
+
+#undef GSPL_OSF_PAIR
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Verify that all elements of a grid are within range [0, 1].
+ *
+ * Used to verify that \c ReflectionForm Grids are within the expected range.
+ */
+bool unity(inp::Grid const& grid)
+{
+    return std::any_of(grid.y.begin(), grid.y.end(), [](real_type const& val) {
```

Comment:
The name isn't right ; and it should be checking all the points not "any". Maybe:
```suggestion
bool is_fractional(inp::Grid const& grid)
{
    return std::all_of(grid.y.begin(), grid.y.end(), [](real_type const& val) {
```

Reply:
Maybe `is_probability`?